spec/datadog/tracing/span_operation_spec.rb

```diff
@@ -995,6 +995,86 @@
       end
     end
   end
+
+  describe '#record_exception' do
+    subject(:record_exception) do
+      begin
+        raise error
+      rescue => e
+        @error_object = Datadog::Core::Error.build_from(e)
```

Comment:
This should already been handled by the method `record_exception`.

Reply:
@error_object is not used in `record_exception`. I did that because this is the only way of having the exact same stacktrace as the span_vent.

Reply:
You should be able to provide the backtrace during exception initialization with `#set_backtrace` 

```
let(:error) { StandardError.new('test error') }
```